lib/game.rb

```diff
@@ -0,0 +1,87 @@
+class Game
+
+  attr_accessor :player_1, :player_2, :board
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],#horizontal
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],#vertical
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],#diagonal
+    [2, 4, 6],
+  ]
+
+  def current_player
+    if @board.turn_count % 2 == 0
+      @player_1
+    else
+      @player_2
+    end
+  end
+
+  def over?
+    if won? || draw?
+      true
+    else
+      false
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combination|
+      @board.cells[combination[0]] == @board.cells[combination[1]] &&
+      @board.cells[combination[1]] == @board.cells[combination[2]] &&
+      @board.taken?(combination[0] + 1)
+  end
+
+  end
+
+  def draw?
+    if @board.full? && !won?
+      true
+    else
+      false
+    end
+  end
+
+  def winner
+    WIN_COMBINATIONS.each do |combination|
```

Comment:
Consider using your `won?` method as a helper to return the winner of the game. 

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.